core/src/main/java/com/linecorp/armeria/common/stream/StreamWriter.java

```diff
@@ -33,6 +35,31 @@
      */
     boolean isOpen();
 
+    /**
+     * Writes the specified object to the {@link StreamMessage}. The written object will be transferred to the
+     * {@link Subscriber}.
+     *
+     * @throws AbortedStreamException if the stream was already closed
+     * @throws IllegalArgumentException if the publication of the specified object has been rejected
+     */
+    default void write(T o) {
+        if (!tryWrite(o)) {
+            throw AbortedStreamException.get();
```

Comment:
How about using `new IllegalStateException("stream closed")`? `AbortedStreamException` is specifically for the notification of `abort()`.

Reply:
Sure - done